openedx/core/lib/xblock_builtin/xblock_discussion/static/js/discussion_classes.js

```diff
@@ -0,0 +1,37 @@
+// Add RequireJS definitions for each discussion class
+
+var discussionClasses = [
+    ['Discussion', 'common/js/discussion/discussion'],
+    ['DiscussionModuleView', 'common/js/discussion/discussion_module_view'],
+    ['DiscussionThreadView', 'common/js/discussion/views/discussion_thread_view'],
+    ['DiscussionThreadListView', 'common/js/discussion/views/discussion_thread_list_view'],
+    ['DiscussionThreadProfileView', 'common/js/discussion/views/discussion_thread_profile_view'],
+    ['DiscussionUtil', 'common/js/discussion/utils'],
+    ['NewPostView', 'common/js/discussion/views/new_post_view']
+];
+
+var defineDiscussionClasses = function(define) {
+    'use strict';
+    discussionClasses.forEach(function(discussionClass) {
+        define(
+            discussionClass[1],
+            [],
+            function() {
+                return window[discussionClass[0]];
+            }
+        );
+    });
+};
+
+if (typeof RequireJS === 'undefined') {
+    var vendorScript = document.createElement("script");
+    vendorScript.onload = function() {
+        'use strict';
+        defineDiscussionClasses(define);
```

Comment:
@ziafazal This is a bit difficult for me to debug, since I can't reproduce the problem. Could you please tell me what your exact setup is (in terms of commits and anything else that you think would be useful to know)?

BTW,  I didn't mean to suggest in my previous [comment](https://github.com/edx-solutions/edx-platform/pull/744#issuecomment-249863368) that reverting https://github.com/edx-solutions/edx-platform/pull/744/commits/0a2af9c33db5d3c2e919e51b0399d03a1773c885 would fix the problem on your end (if that's the impression you got). What I meant was that reverting the commit might help with debugging the issue. When you load a unit with an instance of `DiscussionCourseXBlock` in Apros, do you get the following output (or do some of these messages appear multiple times)?

```
URI is not defined. Loading it now ...
RequireJS is not defined. Loading it now ...
edx namespace is not defined. Loading it now ...
```